reference/host-crd.md

```diff
@@ -1,8 +1,167 @@
+* ACME challenges with prefix `/.well-known/acme-challenge/` are always forced to be considered insecure, since they are not supposed to arrive over HTTPS.
+* Ambassador Edge Stack provides native handling of ACME challenges. If you are using this support, Ambassador will automatically arrange for insecure ACME challenges to be handled correctly. If you are handling ACME yourself - as you must when running Ambassador Open Source - you will need to supply appropriate Host resources and Mappings to correctly direct ACME challenges to your ACME challenge handler.
+* In some cases - like the split L4 scenario below - you will need to have a listener on both port 8443 and port 8080, but you will need to not have a TLSContext. The way to do this is to set the service port for Ambassador to 8443 using the Ambassador module, then explicitly set an insecure additionalPort of 8080 in your Host resource.
+
+## Use Cases and Examples
+
+1. The most common use case involves primarily using HTTPS, but redirecting HTTP to HTTPS:
+
+  ```yaml
+  apiVersion: getambassador.io/v2
+  kind: Host
+  metadata:
+    name: minimal-host
+  spec:
+    hostname: host.example.com
+    acmeProvider: <as needed>
+  requestPolicy:
+    insecure:
+      action: redirect
+  ```
+
+  Since this is the default, the `requestPolicy` element could also simply be dropped.
+
+2. HTTPS-only, TLS terminated at Ambassador:
+
+  ```yaml
+  apiVersion: getambassador.io/v2
+  kind: Host
+  metadata:
+    name: minimal-host
+  spec:
+    hostname: host.example.com
+    acmeProvider: <as needed>
+  requestPolicy:
+    insecure:
+      action: reject
+  ```
+
+  We need to make sure to set the `acmeProvider` appropriately for Ambassador to manage certificates for both of the previous cases.
+
+3. HTTPS-only, TLS terminated at a load balancer:
+
+  ```yaml
+  apiVersion: getambassador.io/v2
+  kind: Host
+  metadata:
+    name: minimal-host
+  spec:
+    hostname: host.example.com
+    acmeProvider:
+      authoriry: none
+
+  requestPolicy:
+    insecure:
+      action: reject
+  ```
+
+  This configuration relies on the load balancer to set `X-Forwarded-Proto` correctly, so that Ambassador can tell insecure requests from secure requests. We also need to explicitly set the `acmeProvider` to none, so that Ambassador doesn’t try to do certificate management when it shouldn’t.
+
+4. HTTP-only:
+
+  ```yaml
+  apiVersion: getambassador.io/v2
+  kind: Host
+  metadata:
+    name: minimal-host
+  spec:
+    hostname: host.example.com
+    acmeProvider:
+      authority: none
+  requestPolicy:
+    insecure:
+      action: route
+  ```
+
+  In this case, the Host resource explicitly requests no ACME handling, then states that insecure requests must be routed instead of redirected.
+
+### Split L4 Load Balancer
```

Comment:
The split L4 load balancer is a use case -- either it should be part of the numbered list, or all the other elements should be H3's.